# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCRegMgmt-inl.h
# Commit: fb02656bb788
# Full Hash: fb02656bb788b8111d7159b1691094e429e7fc4f
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 6
# Description:
#   Bug 1727084 - Memory64 - Non-atomic-RMW baseline operations. r=yury
#   
#   Implement scalar, SIMD, and atomic loads and stores for the baseline
#   compiler for x64, x86, arm64, and arm32.  Atomic RMW operations and
#   wait/notify are in the next patch.
# ==============================================================================

diff -r 13ee9674ee35 -r fb02656bb788 js/src/wasm/WasmBCRegMgmt-inl.h
--- a/js/src/wasm/WasmBCRegMgmt-inl.h	Mon Oct 18 09:58:14 2021 +0000
+++ b/js/src/wasm/WasmBCRegMgmt-inl.h	Mon Oct 18 09:58:15 2021 +0000
@@ -212,6 +212,13 @@
 
 RegI32 BaseCompiler::fromI64(RegI64 r) { return RegI32(lowPart(r)); }
 
+RegI32 BaseCompiler::maybeFromI64(RegI64 r) {
+  if (!r.isValid()) {
+    return RegI32::Invalid();
+  }
+  return fromI64(r);
+}
+
 #ifdef JS_PUNBOX64
 RegI64 BaseCompiler::fromI32(RegI32 r) { return RegI64(Register64(r)); }
 #endif